sys/kern/link_elf.c

```diff
@@ -1792,13 +1792,44 @@ link_elf_lookup_debug_symbol_ctf(linker_file_t lf, const char *name,
 	return (i < ef->ddbsymcnt ? link_elf_ctf_get_ddb(lf, lc) : ENOENT);
 }
 
+static void
+link_elf_ifunc_symbol_value(linker_file_t lf, caddr_t *valp, size_t *sizep)
+{
+	c_linker_sym_t sym;
+	elf_file_t ef;
+	const Elf_Sym *es;
+	caddr_t val;
+	long off;
+
+	val = *valp;
+	ef = (elf_file_t)lf;
+
+	/* Provide the value and size of the target symbol, if available. */
+	val = ((caddr_t (*)(void))val)();
+	if (link_elf_search_symbol(lf, (ptraddr_t)val, &sym, &off) == 0 &&
+	    off == 0) {
+		es = (const Elf_Sym *)sym;
+#ifdef __CHERI_PURE_CAPABILITY__
+		*valp = make_capability(es,
```

Comment:
Why are we rederiving here? This should be identical to val unless there are bugs?

Reply:
You're right, it's sufficient to set `*valp = val` instead.